algorithms/gr-bfsrh/cpp/BFS.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <list>
+
+using namespace std;
+
+class Graph {
+  int numVertices;
+  list<int>* adjLists;
+  bool* visited;
+
+   public:
+  Graph(int vertices);
+  void addEdge(int src, int dest);
+  void BFS(int startVertex);
+};
+
+// Create a graph with given vertices,
+// and maintain an adjacency list
+Graph::Graph(int vertices) {
+  numVertices = vertices;
+  adjLists = new list<int>[vertices];
+}
+
+// Add edges to the graph
+void Graph::addEdge(int src, int dest) {
+  adjLists[src].push_back(dest);
+  adjLists[dest].push_back(src);
+}
+
+// BFS algorithm
+void Graph::BFS(int startVertex) {
+  visited = new bool[numVertices];

```

Comment:
Codacy found an issue: [Possible leak in public function. The pointer 'visited' is not deallocated before it is allocated.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6318755)